### PR TITLE
add hacking with yearn guide

### LINKED ISSUE #195: Add Section | Hacking on Yearn
include all material here

https://hackmd.io/BP6CjvX-T_-CbfsksR7FFQ?both

needs to be separate section

### PR DESCRIPTION
**Pull request by MarcoWorms**
closes #195 

also formats the sidebar config file into a more readable form (every line is a file now)

### CODE REVIEW

**Comment by MarcoWorms** [comment]
link linter says that a link is broken but it isn't .-.

File: docs/developers/v2/hacking-with-yearn.md [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+This page is for the hackers and buidlers who are currently building, or planning to build, something cool with Yearn Finance. This guide  details all the info you need to start hacking!
+
+## Contents
+- Vault and Smart Contract Integrations
+- Front-end and SDK Integrations
+- Strategy Development
+- Data Analysis
+- Inspiration
+- Examples
+- Expectations
```
Comment (saltyfacu) [inline docs/developers/v2/hacking-with-yearn.md:12]: none of the articles have a ToC. I don't think we need it here. You can describe very very briefly what they are going to find in the article in the intro.

File: docs/developers/v2/hacking-with-yearn.md [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+- [Foundry](https://github.com/nicolasgarcia214/damn-vulnerable-defi-foundry)
+
+# Yearn Vaults & Smart Contract Integrations
+Yearn’s core product is the Vault (also known as yVault) — a set-it-and-forget-it yield aggregator running on top of DeFi’s yield-generating protocols like Compound, Aave, Curve, and Convex. Most of the core functionality such as deposit, withdraw, borrow, repay, or flashloan occurs in the XX contract ().
+For a breakdown of V2 features see the contract overview or dive even deeper with the V2 Technical Paper.
+
+## Getting Started
+
+#### Starter Kit
+####
```
Comment (saltyfacu) [inline docs/developers/v2/hacking-with-yearn.md:30]: is this necessary? seems like an empty title

File: docs/developers/v2/hacking-with-yearn.md [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+https://github.com/storming0x/ystarter-foundry-kit
+
+What you'll find:
+- Basic example Solidity Smart Contracts for integrating with Yearn Vaults for both V2 and ERC4626 interfaces.
+- ERC4626 adapter to wrap yearn vaults.
+- Example Contracts
+- Sample test suite.
+
+
+Yearn Vaults
```
Comment (saltyfacu) [inline docs/developers/v2/hacking-with-yearn.md:42]: is this a subtitle? a title? a link? a bullet?

File: docs/developers/v2/hacking-with-yearn.md [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+
+[This repository](https://github.com/yearn/yearn-vaults#readme) includes the set of smart contracts that are used for the Yearn Vaults. It contains the requirements, code, deployment scripts, and tests necessary for the core protocol, including a inheritable template strategy for use with Solidity-based strategies that interact with Yearn Vaults. These contracts are used to create a simple way to generate high risk-adjusted returns for depositors of various assets via best- in-class lending protocols, liquidity pools, and community-made yield farming strategies on Ethereum.
+
+
+####
```
Comment (saltyfacu) [inline docs/developers/v2/hacking-with-yearn.md:47]: another empty title

File: docs/developers/v2/hacking-with-yearn.md [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+
+Yearn Vaults
+
+[This repository](https://github.com/yearn/yearn-vaults#readme) includes the set of smart contracts that are used for the Yearn Vaults. It contains the requirements, code, deployment scripts, and tests necessary for the core protocol, including a inheritable template strategy for use with Solidity-based strategies that interact with Yearn Vaults. These contracts are used to create a simple way to generate high risk-adjusted returns for depositors of various assets via best- in-class lending protocols, liquidity pools, and community-made yield farming strategies on Ethereum.
+
+
+#### 
+## Testing
+
+### **Introduction**
```
Comment (saltyfacu) [inline docs/developers/v2/hacking-with-yearn.md:50]: i don't think that a title with ** ** is a good idea. Let's keep the styling clean

File: docs/developers/v2/hacking-with-yearn.md [hunk 6]
```diff
@@ -0,0 +1,237 @@
...
+Testing is possible through Ethereum Rinkeby  
+
+**Steps**:
+- deploy your vault with your own token
+- mint your own tokens 
+- hack!
+
+
+### What do I need for testing?
+###
```
Comment (saltyfacu) [inline docs/developers/v2/hacking-with-yearn.md:68]: another empty title

File: docs/developers/v2/hacking-with-yearn.md [hunk 7]
```diff
@@ -0,0 +1,237 @@
...
+Make sure to switch your wallet provider to the appropriate testnet network in order to use the protocol testnet without incurring costs on the mainnet network. 
+ 
+Testnet networks are different environments from mainnet and simulate real use cases for development, testing or production. The assets on a testnet are not “real,” meaning that they cannot be withdrawn to your wallet for any real economic value. 
+ 
+Testing is possible through Ethereum Rinkeby  
+
+**Steps**:
+- deploy your vault with your own token
+- mint your own tokens 
+- hack!
```
Comment (saltyfacu) [inline docs/developers/v2/hacking-with-yearn.md:64]: the bullets above start with caps, these don't. why?

File: docs/developers/v2/hacking-with-yearn.md [hunk 8]
```diff
@@ -0,0 +1,237 @@
...
+When accessing each testnet, you will need to use its own native asset token. 
+ 
+Steps to test: 
+
+- Go to https://hack.yearn.finance/
+- Switch to the testnet you wish to utilize over your wallet provider 
+- Make sure to have the native asset for the specific network 
+- Get some tokens from the test client faucet  
+Ethereum Rinkeby: https://faucet.rinkeby.io/  
+
```
Comment (saltyfacu) [inline docs/developers/v2/hacking-with-yearn.md:78]: no bold on "Steps to test" as in the other bullets, there's a space between the bullets that was not in the previous ones

File: docs/developers/v2/hacking-with-yearn.md [hunk 9]
```diff
@@ -0,0 +1,237 @@
...
+
+Make sure to select the correct market in the top right corner. You can find this towards the left side of the address. Select the token you need and submit the faucet transaction but be mindful to make sure that you have certain amount of the native asset of the testnet you wish to transact on.
+
+#### How do I select a supported testnet?
+####
```
Comment (saltyfacu) [inline docs/developers/v2/hacking-with-yearn.md:83]: empty title...

File: docs/developers/v2/hacking-with-yearn.md [hunk 10]
```diff
@@ -0,0 +1,237 @@
...
+invest the yield in more risky stuff or directional bets
+paying bills and subscriptions
+3. Create a metavault on top of multiple vaults. One idea that comes to my mind is having a USD metavault that deposits into multiple dollar pegged vaults based on some strategy and abstracts the complexity from the user.
+4. Build a UI that accepts deposits and allows you to move assets across chains.
+5. Use the vaults as a place to earn interest on funds that are not currently actively being used by the user.
+
+
+
+
+
```
Comment (saltyfacu) [inline docs/developers/v2/hacking-with-yearn.md:207]: all those spaces are not going to give you more space during the render

**Review by saltyfacu — APPROVED** [review]
let's merge and continue polishing

**Pull request merged**